azure-pipelines/end-to-end-tests-dir/add.ps1

```diff
@@ -9,18 +9,59 @@ New-Item -Path $manifestDir -ItemType Directory
 New-Item -Path $manifestPath -ItemType File `
     -Value (ConvertTo-Json -Depth 5 -InputObject $vcpkgJson)
 
-Run-Vcpkg add port zlib @manifestDirArgs
+Run-Vcpkg add port sqlite3 @manifestDirArgs
 Throw-IfFailed
 
 $expected = @"
 {
   "dependencies": [
-    "zlib"
+    "sqlite3"
   ]
 }
 "@
 
 $actual = (Get-Content -Path $manifestPath -Raw).TrimEnd()
 if ($expected -ne $actual) {
-    throw "Add port didn't add zlib dependency correctly.`nExpected: $expected`nActual:$actual"
+    throw "Add port didn't add sqlite3 dependency correctly.`nExpected: $expected`nActual:$actual"
+}
+
+Run-Vcpkg add port "sqlite3[core]" @manifestDirArgs
+Throw-IfFailed
+
+$expected = @"
+{
+  "dependencies": [
+    {
+      "name": "sqlite3",
+      "default-features": false
+    }
+  ]
+}
+"@
+
+$actual = (Get-Content -Path $manifestPath -Raw).TrimEnd()
+if ($expected -ne $actual) {
+    throw "Add port didn't add sqlite3[core] dependency correctly.`nExpected: $expected`nActual:$actual"
+}
+
+Run-Vcpkg add port "sqlite3[zlib]" @manifestDirArgs
+Throw-IfFailed
+
+$expected = @"
+{
+  "dependencies": [
+    {
+      "name": "sqlite3",
+      "default-features": false,
```

Comment:
I think this should be clarified that this is still here from the previous run, but it would also be better to have an independent test with `[core,somethingelse]`.

Reply:
On second thought I think this should make default features be true because that's what `sqlite3[zlib]` requests.

@ras0219-msft also points out that we should have tests for multiple references to the same name

Reply:
I now have the test `vcpkg add port "sqlite3" "sqlite3[core]"`